src/flib/pionfput_mod.F90.in

```diff
@@ -159,13 +159,17 @@ contains
     call t_startf("PIO:put_var1_{TYPE}")
 #endif
     clen = size(index)
-    allocate(cindex(clen))
-    do i=1,clen
-       cindex(i) = index(clen-i+1)-1
-    enddo
-
-    ierr = PIOc_put_var1_{NCTYPE} (file%fh, varid-1, cindex, ival)
-    deallocate(cindex)
+    if (clen > 0) then
+        allocate(cindex(clen))
+        do i=1,clen
+            cindex(i) = index(clen-i+1)-1
+        enddo
+
+        ierr = PIOc_put_var1_{NCTYPE} (file%fh, varid-1, cindex, ival)
+        deallocate(cindex)
+    else
+        print *, 'Warning, clen <= 0 in put_var1_{TYPE} ', __FILE__, __LINE__
```

Comment:
- Can you change `__FILE__` to `__PIO_FILE__` ? 
- Can you also add a more informative message to the user (similar to the ones in the C library APIs - a message that makes more sense to the user)?

The same comment on the changes below

Reply:
OK, I will update this PR with your suggestions.